### PR TITLE
fix: delete unnecessary state update

### PR DESCRIPTION
**Pull request by solittlework**
Updating the template state within the onChange function might be unnecessary. We only need to update the state and sync with the remote when we click the save button.

### CODE REVIEW

File: apps/tenant-management-webapp/src/app/pages/admin/services/pdf/templates/templates.tsx [hunk 1]
```diff
@@ -122,10 +122,6 @@ export const PdfTemplates: FunctionComponent<PdfTemplatesProps> = ({ openAddTemp
       setFooterPreview(
         generateMessage(template, { data: currentTemplate, serviceUrl: webappUrl, today: new Date().toDateString() })
       );
-      if (currentTemplate) {
-        currentTemplate.footer = footer;
-      }
-      setCurrentTemplate(currentTemplate);
```
Comment (jonathanweyermann) [inline apps/tenant-management-webapp/src/app/pages/admin/services/pdf/templates/templates.tsx:128]: removing this loses changes between tabs if you haven't saved it yet

File: apps/tenant-management-webapp/src/app/pages/admin/services/pdf/templates/templates.tsx [hunk 2]
```diff
@@ -141,10 +137,6 @@ export const PdfTemplates: FunctionComponent<PdfTemplatesProps> = ({ openAddTemp
       setHeaderPreview(
         generateMessage(template, { data: currentTemplate, serviceUrl: webappUrl, today: new Date().toDateString() })
       );
-      if (currentTemplate) {
-        currentTemplate.header = header;
-      }
-      setCurrentTemplate(currentTemplate);
```
Comment (jonathanweyermann) [inline apps/tenant-management-webapp/src/app/pages/admin/services/pdf/templates/templates.tsx:147]: removing this loses changes between tabs if you haven't saved it yet

File: apps/tenant-management-webapp/src/app/pages/admin/services/pdf/templates/templates.tsx [hunk 3]
```diff
@@ -160,10 +152,6 @@ export const PdfTemplates: FunctionComponent<PdfTemplatesProps> = ({ openAddTemp
       setBodyPreview(
         generateMessage(template, { data: currentTemplate, serviceUrl: webappUrl, today: new Date().toDateString() })
       );
-      if (currentTemplate) {
-        currentTemplate.template = body;
-      }
-      setCurrentTemplate(currentTemplate);
```
Comment (jonathanweyermann) [inline apps/tenant-management-webapp/src/app/pages/admin/services/pdf/templates/templates.tsx:166]: removing this loses changes between tabs if you haven't saved it yet

**Review by jonathanweyermann — CHANGES_REQUESTED** [review]
see comments

File: apps/tenant-management-webapp/src/app/pages/admin/services/pdf/templates/templates.tsx (hunk 1 above)
Comment (solittlework) [reply apps/tenant-management-webapp/src/app/pages/admin/services/pdf/templates/templates.tsx:128]: Thanks, Jonathan! I am going to see whether I can save it when we click the tab.

**Comment by jonathanweyermann** [comment]
Uploading Alberta Digital Service Platform - Google Chrome 2022-09-27 16-00-51.mp4…

File: apps/tenant-management-webapp/src/app/pages/admin/services/pdf/templates/templates.tsx (hunk 1 above)
Comment (solittlework) [reply apps/tenant-management-webapp/src/app/pages/admin/services/pdf/templates/templates.tsx:128]: Hi @jonathanweyermann, thank you for the review. I've updated the PR and a copy of the template state will be updated when we click the tab. The update will resolve all of the issues here. Please help me review it again. Thanks!

**Review by jonathanweyermann — APPROVED** [review]

**Review by jonathanweyermann — CHANGES_REQUESTED** [review]
found something else

**Comment by jonathanweyermann** [comment]
Not sure what happened, but this the footer input is totally broken and it doesn't happen in UAT


https://user-images.githubusercontent.com/11400938/192820053-7e3b5109-2bf9-4333-bd91-25bf032c3bfe.mp4

**Review by jonathanweyermann — APPROVED** [review]

**Pull request merged**